specs/patterns.md

```diff
@@ -402,4 +402,149 @@ In this case you want to specify that A depends on B and C -- and that "installi
+
+```javascript
+// with data and a schema accessible via the local filesystem
+{
+  "profile": "document-data-resource",
+  "name": "resource-name",
+  "path": [ "resource-path.json" ],
+  "schema": "jsonschema.json"
+}
+
+// with data accessible via http
+{
+  "profile": "document-data-resource",
+  "name": "resource-name",
+  "path": [ "http://example.com/resource-path.json" ],
+  "schema": "http://example.com/jsonschema.json"
+}
+```
+
+A minimal Document Data Resource example using the data property to inline data looks as follows.
+
+```javascript
+{
+  "profile": "document-data-resource",
+  "name": "resource-name",
+  "data": {
+    "id": 1,
+    "first_name": "Louise"
+  },
+  "schema": {
+    "type": "object",
+    "required": [
+      "id"
+    ],
+    "properties": {
+      "id": {
+        "type": "integer"
+      },
+      "first_name": {
+        "type": "string"
+      }
+    }
+  }
+}
+```
+
+A comprehensive JSON Data Resource example with all required, recommended and optional properties looks as follows.
+
+```javascript
+{
+  "profile": "document-data-resource",
+  "name": "solar-system",
+  "path": "http://example.com/solar-system.json",
+  "title": "The Solar System",
+  "description": "My favourite data about the solar system.",
+  "format": "json",
+  "mediatype": "application/json",
+  "encoding": "utf-8",
+  "bytes": 1,
+  "hash": "",
+  "schema": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "required": [
+      "id"
+    ],
+    "properties": {
+      "id": {
+        "type": "integer"
+      },
+      "name": {
+        "type": "string"
+      }
+      "description": {
+        "type": "string"
+      }
+    }
+  },
+  "sources": [{
+    "title": "The Solar System - 2001",
+    "path": "http://example.com/solar-system-2001.json",
+    "email": ""
+  }],
+  "licenses": [{
+    "name": "CC-BY-4.0",
+    "title": "Creative Commons Attribution 4.0",
+    "path": "https://creativecommons.org/licenses/by/4.0/"
+  }]
+}
+```
+
+
+### Specification
+
+A Document Data Resource MUST be a [Data Resource][dr], that is it MUST conform to the [Data Resource specification][dr].
+
+In addition:
+
+* The Data Resource `schema` property MUST follow the [JSON Schema](https://json-schema.org/) specification,
+  either as a JSON object directly under the property, or a string referencing another
+  JSON document containing the JSON Schema
+- There `MUST` be a `profile` property with the value `document-data-resource`
+* The data the Data Resource describes MUST, if non-inline, be a JSON file
+
+
+### JSON file requirements
+
+When `"format": "json"`, files must strictly follow the [JSON specification](https://www.json.org/). Some implementations `MAY` support `"format": "jsonc"`, allowing for non-standard single line and block comments (`//` and `/* */` respectively).
+
+### Implementations
+
+None known.
+
+
+## Data Package Resources
```

Comment:
This heading (and subsequent 2 paras) does is not very clear to me and may even be confusing as we already have a definition of Data Package Resources. I think you could just move Data Catalog straight up without having it beneath this heading.